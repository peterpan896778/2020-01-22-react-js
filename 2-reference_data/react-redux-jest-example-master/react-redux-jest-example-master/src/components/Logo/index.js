import React from 'react';
import styled from 'styled-components';

const StyledLogo = styled.img`
  animation: App-logo-spin infinite 20s linear;
  height: 80px;

  @keyframes App-logo-spin {
    from { transform: rotate(0deg); }
    to { transform: rotate(360deg); }
  };
`;

const Logo = ({ ...props }) => <StyledLogo {...props} />;
export default Logo;
