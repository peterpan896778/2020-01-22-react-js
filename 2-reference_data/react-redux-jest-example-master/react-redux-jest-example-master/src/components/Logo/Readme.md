If you want to use a Logo you have to import from `src/components`:
```javascript
import { Logo } from 'src/components';
```

Logo:

    <Logo />
