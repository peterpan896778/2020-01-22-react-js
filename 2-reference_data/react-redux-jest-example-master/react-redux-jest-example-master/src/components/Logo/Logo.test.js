import React from 'react';
import { mount, shallow } from 'enzyme';
import logo from 'src/assets/logo.svg';

import Logo from '.';

test('<Logo /> mounts without crashing', () => {
  const tree = mount(
    <Logo src={logo} />,
  );
  expect(tree).toMatchSnapshot();
});

test('<Logo /> renders without crashing', () => {
  const tree = shallow(
    <Logo src={logo} />,
  );
  expect(tree).toMatchSnapshot();
});
