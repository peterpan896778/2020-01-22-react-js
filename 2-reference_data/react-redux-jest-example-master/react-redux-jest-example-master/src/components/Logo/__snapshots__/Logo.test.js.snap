// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Logo /> mounts without crashing 1`] = `
<Logo
  src="logo.svg"
>
  <styled.img
    src="logo.svg"
  >
    <img
      className="sc-bdVaJa kzBKTv"
      src="logo.svg"
    />
  </styled.img>
</Logo>
`;

exports[`<Logo /> renders without crashing 1`] = `
<styled.img
  src="logo.svg"
/>
`;
